src/components/Header.tsx

```diff
@@ -0,0 +1,71 @@
+import { Component } from 'react';
+import SearchBar from './SearchBar.tsx';
+import { getPokemonList, type Pokemon } from '../services/api.service.tsx';
+import '../styles/Header.scss';
+import LogoImage from '../assets/logo.png';
+import {
+  LocalStorageService,
+  STORAGE_KEYS,
+} from '../services/local-storage.service.tsx';
+
+interface HeaderProps {
+  setPokemons: (data: Pokemon[]) => void;
+  setIsLoading: (data: boolean) => void;
+}
+
+interface HeaderState {
+  searchValue: string;
+}
+
+class Header extends Component<HeaderProps, HeaderState> {
+  constructor(props: HeaderProps) {
+    super(props);
+    const savedSearch =
+      LocalStorageService.getSearchTerm(STORAGE_KEYS.searchTerm) || '';
+    this.state = {
+      searchValue: savedSearch,
+    };
+    if (savedSearch) {
+      this.handleSearch(savedSearch);
+    }
+  }
+
+  handleSearch = (value: string): void => {
+    const { setPokemons, setIsLoading } = this.props;
+
+    this.setState({ searchValue: value });
+    if (value) {
+      LocalStorageService.setSearchTerm(STORAGE_KEYS.searchTerm, value);
+    }
+
```

Comment:
and here change the logic
```suggestion
    value = value.trim();

    if (value) {
      this.setState({ searchValue: value });
      LocalStorageService.setSearchTerm(STORAGE_KEYS.searchTerm, value);
    } else {
      LocalStorageService.clearSearchTerm(STORAGE_KEYS.searchTerm);
      return;
    }

    const { setPokemons, setIsLoading } = this.props;
```